src/utils/spfx.ts

```diff
@@ -14,5 +14,38 @@ export const spfx = {
       typeof project.yoRcJson['@microsoft/generator-sharepoint'] !== 'undefined' &&
       project.yoRcJson["@microsoft/generator-sharepoint"].framework === 'knockout') ||
       typeof project.packageJson?.dependencies?.['knockout'] !== 'undefined';
+  },
+
+  getHighestNodeVersion(versionRange: string): string {
+    const ranges = versionRange.split('||').map(r => r.trim());
+
+    const versions = ranges.map(range => {
+      if (range.includes('<')) {
+        const upperBound = range.split('<')[1].trim().split(' ')[0];
```

Comment:
At this line, we expect that there is a space between the `<` sign and the version number, which shouldn't be the case.